### ISSUE TITLE
Reproduce best_mle_temp_rlm_score

### ISSUE THREAD

**Issue by anonymous1100** [pclucas14/GansFallingShort#8]
I ran score_models.py to evaluate the lm and rlm, loading the trained_models/news/word/best_mle/models/gen80th model. But the rlm results I got were worse than those in your best_mle_temp_rlm_score.csv. 
Are the parameter settings during rlm training incorrect? I used the parameter settings of get_rlm_args in your code and args.mle_epochs=80. 
I have measured the rlm when alpha=1.0 and the result is rlm=4.02, which is much worse than 3.991.
Thanks.

**Comment by pclucas14** [comment]
Hi, 

what was the corresponding LM value for alpha == 1. ? RLM value of 4.02 vs 3.99 is actually a very small difference, which is most likely attributed to the stochasticity of the sampling procedure. 

Best, 
Lucas

**Comment by anonymous1100** [comment]
Thanks Reply,
When alpha = 1.0, LM = 3.6276, the same result as best_mle_temp_lm_score.csv. I also tested the LM at alpha = 1.1, 1.3, 1.5, and their results are the same as in best_mle_temp_lm_score.csv, no problem at all.
I think the LM results are basically the same, but why the rlm results are worse than the paper. At alpha = 1.1, 1.3, 1.5, their rlm is worse than 0.02 in the paper.

**Comment by pclucas14** [comment]
could you post your lm / rlm results for alpha=1.1, 1.3, 1.5 ?

**Comment by anonymous1100** [comment]
alpha = 1.1, lm = 3.201, rlm = 4.139;
alpha = 1.3, lm = 2.637, rlm = 4.49;
alpha = 1.5, lm = 2.301, rlm = 4.928;
The LM result is the same as best_mle_temp_rlm_score.csv, but the RLM result is quite different. Due to insufficient time, here args.mle_epochs = 30.
I wonder if I have a problem with my hyperparameter settings? I am using the parameters of get_rlm_args () in your code, only args.mle_epochs = 30 or 80

**Comment by pclucas14** [comment]
I see, these results are indeed slightly lower than the reported ones. Let me check older commits to see what `mle_epochs` value was used 

-Lucas

**Comment by pclucas14** [comment]
There was indeed a flag I forgot to add in the `rlm_args` defaults (I forgot to commit the file when I cleaned up the repo for the ICLR release). Can you try now with the latest changes ?

**Comment by anonymous1100** [comment]
OK, I can.

**Comment by anonymous1100** [comment]
Thank you very much for your help, I will continue to try now.
I have been following your work since last year, and congratulations for your paper published on ICLR.

**Comment by anonymous1100** [comment]
Excuse me,&nbsp; there is still a problem here.&nbsp;
The rlm training process does not involve the args.old_model parameter.

---Original---

**Comment by pclucas14** [comment]
Yes, you are absolutely right. I forgot to push the changes in `main.py`. I've pushed all my local changes, so hopefully now everything works. 

I ran `python score_models.py --model_path trained_models/news/word/best_mle --decoder temp` for `alpha=1.5` for which I obtained `LM=2.30` and `RLM=4.83`.  This seems to be exactly what I have in the `.csv`.  The RLM calculation only took 8 epochs, so it shouldn't be too long to check. 

Thanks for flagging this!
-Lucas
